File: 240-search-a-2d-matrix-ii/240-search-a-2d-matrix-ii.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        
        for i in range(len(matrix)):
            row = matrix[i]

            if row[-1] < target:
                continue
            
            if row[0] > target:
                break
            
            l,r = 0,len(row) - 1

            while l <= r:
                mid = (l + r)//2

                if row[mid] < target:
                    l = mid + 1
                
                elif row[mid] > target:
                    r = mid - 1
                
                else:
                    return True
        
        return False

File: 240-search-a-2d-matrix-ii/240-search-a-2d-matrix-ii.js
/**
 * @param {number[][]} matrix
 * @param {number} target
 * @return {boolean}
 */
var searchMatrix = function(matrix, target) {
    let r = matrix.length - 1;
    let c = 0;
    
    while(r >= 0 && c < matrix[0].length){
        if(matrix[r][c] > target) r -= 1;
        else if(matrix[r][c] < target) c += 1;
        else return true;
    }
    return false;
};